cpp/src/lance/arrow/utils.h

```diff
@@ -31,6 +32,15 @@ ::arrow::Result<std::shared_ptr<::arrow::RecordBatch>> MergeRecordBatches(
     const std::shared_ptr<::arrow::RecordBatch>& rhs,
     ::arrow::MemoryPool* pool = ::arrow::default_memory_pool());
 
+/// Merge a list of record batches into one.
```

Comment:
add a quick note on what merge means here?
is it horizontal or vertical concat?

Reply:
Sure, will do.